src/ClientRuntime/Microsoft.Rest.ClientRuntime/ServiceClient.cs

```diff
@@ -212,6 +215,19 @@ protected ServiceClient()
         /// <summary>
         /// Initializes a new instance of the ServiceClient class.
         /// </summary>
+        /// <param name="httpClient">HttpClient</param>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage(
+            "Microsoft.Reliability",
+            "CA2000:Dispose objects before losing scope",
+            Justification = "The created objects should be disposed on caller's side")]
+        protected ServiceClient(HttpClient httpClient)
+        {
+            InitializeHttpClient(httpClient, CreateRootHandler());
```

Comment:
I'm not sure - some properties of HttpClient are unsettable after the client is used for a request, this is why the scenario of reusing an http client after a request should be covered in a test